cpp/ql/lib/semmle/code/cpp/ir/dataflow/internal/DataFlowPrivate.qll

```diff
@@ -239,28 +239,81 @@ OutNode getAnOutNode(DataFlowCall call, ReturnKind kind) {
     )
 }
 
+private predicate writesToGlobal(StoreInstruction store, GlobalOrNamespaceVariable v) {
+  v = store.getResultAddress().(VariableAddressInstruction).getAstVariable()
+}
+
+private predicate loadsFromGlobal(LoadInstruction load, GlobalOrNamespaceVariable v) {
+  v = load.getSourceAddress().(VariableAddressInstruction).getAstVariable()
+}
+
+/** Holds if `i1` is always executed after `i2` */
+bindingset[i1, i2]
+private predicate instrStrictlyPostDominates(Instruction i1, Instruction i2) {
+  i1.getBlock().strictlyPostDominates(i2.getBlock())
+  or
+  exists(IRBlock block, int index1, int index2 |
+    block = i1.getBlock() and
+    block = i2.getBlock() and
+    block.getInstruction(index1) = i1 and
+    block.getInstruction(index2) = i2 and
+    index1 > index2
+  )
+}
+
+/** Holds if `i1` is always executed before `i2` */
+bindingset[i1, i2]
+private predicate instrStrictlyDominates(Instruction i1, Instruction i2) {
+  i1.getBlock().strictlyDominates(i2.getBlock())
+  or
+  exists(IRBlock block, int index1, int index2 |
+    block = i1.getBlock() and
+    block = i2.getBlock() and
+    block.getInstruction(index1) = i1 and
+    block.getInstruction(index2) = i2 and
+    index1 < index2
+  )
+}
+
+predicate jumpInto(GlobalOrNamespaceVariable v, Node n1, Node n2) {
+  exists(StoreInstruction store |
+    store = n1.asInstruction() and
+    writesToGlobal(store, v) and
+    // Only do a jump step if there is no write that is guarenteed to
+    // overwrite the global variable.
+    not exists(StoreInstruction anotherStore |
+      store != anotherStore and
```

Comment:
Doesn't the "strictly" in `instrStrictlyPostDominates` already give us this?

Reply:
Oh yes, you're totally right 👍. Will fix!

Reply:
Fixed in 7d1bfba1bef7933450afdec7088a21386b6fe729.